### PR TITLE
linting 47 pr

### PR DESCRIPTION
**Pull request by faddat**
This PR lints the sdk 47 work, including tests.  

It also includes #1259 since I think that'll merge into develop_sdk47 first.

### CODE REVIEW

**Comment by alpe** [comment]
Thank you for your help! We are working hard to bring the 47 dev branch to main. If you can help with code reviews, that would be very much appreciated at this stage. 
When 47 is in main, I would be happy to merge the linter fixes on the prod code so that we have this addressed, too. Otherwise we keep fighting merge conflicts that make our work harder. Please bare with us.

**Comment by faddat** [comment]
Linting is part of how I review code.  It's actually what I do before doing that, because it makes the code easier to review and makes the editor automatically highlight small things that could otherwise be missed in review.

File: x/wasm/genesis_test.go [hunk 1]
```diff
@@ -82,7 +82,8 @@ func TestInitGenesis(t *testing.T) {
 	q2 := newData.grpcQueryRouter
 
 	// initialize new app with genstate
-	InitGenesis(newData.ctx, &newData.keeper, *genState)
+	_, err = InitGenesis(newData.ctx, &newData.keeper, *genState)
+	require.NoError(t, err)
```
Comment (alpe) [inline x/wasm/genesis_test.go:86]: 🤔 This is failing in this branch.

**Review by alpe — COMMENTED** [review]
Thanks for the work! 💐 
I added some minor comments. I am wondering now, why the test is failing. It does not look related to your changes.

File: x/wasm/genesis_test.go (hunk 1 above)
Comment (faddat) [reply x/wasm/genesis_test.go:86]: aha... yeah... so, I had... oddities here.

Comment (alpe) [reply x/wasm/genesis_test.go:86]: I found the root cause for the issue in the fuzzer setup. I will do a PR. btw, this is not the line failing

File: .golangci.yml [hunk 2]
```diff
@@ -8,53 +10,59 @@ linters:
...
+    - nolintlint
+    - revive
     - staticcheck
     - stylecheck
     - typecheck
     - unconvert
+    - unparam
     - unused
 
+# TODO: fix the use of deprecated gov style, that'll remove the exclusion of "simtypes" below.
```
[the remark this answers is not in the document]
Comment (faddat) [reply .golangci.yml:32]: makese sense.

File: x/wasm/genesis_test.go (hunk 1 above)
Comment (faddat) [reply x/wasm/genesis_test.go:86]: I'm not surprised, and I wonder if it is maybe related to failures at line 127 in the same file.

**Comment by faddat** [comment]
I've merged #1263 here, as well.

File: x/wasm/genesis_test.go (hunk 1 above)
Comment (faddat) [reply x/wasm/genesis_test.go:86]: ... they were both intermittent.

File: x/wasm/keeper/handler_plugin_encoders_test.go [hunk 3]
```diff
@@ -11,10 +11,10 @@ import (
 	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 	"github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/cosmos/gogoproto/proto" //nolint:staticcheck // we need to use the old protobuf API
```
Comment (alpe) [inline x/wasm/keeper/handler_plugin_encoders_test.go]: Is the `//nolint:staticcheck` still needed? the comment is obsolete

File: x/wasm/keeper/reflect_test.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ import (
 	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
 	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
-	"github.com/golang/protobuf/proto" //nolint:staticcheck // we'd like to use this version of the library
+	"github.com/cosmos/gogoproto/proto" //nolint:staticcheck // we'd like to use this version of the library
```
Comment (alpe) [inline x/wasm/keeper/reflect_test.go]: same question about `//nolint:staticcheck` and note about comment

**Review by alpe — APPROVED** [review]
Minor nit but happy to merge this now. Thanks for pushing this forward!

File: x/wasm/keeper/handler_plugin_encoders_test.go (hunk 3 above)
Comment (alpe) [reply x/wasm/keeper/handler_plugin_encoders_test.go]: This was fixed already. sorry

**Pull request merged**